src/components/EnterpriseApp/data/enterpriseCurationReducer.js

```diff
@@ -43,18 +48,59 @@ export const enterpriseCurationActions = {
     type: ADD_HIGHLIGHT_SET,
     payload,
   }),
+  setEnterpriseHighlightedContents: (payload) => ({
+    type: SET_ENTERPRISE_HIGHLIGHTED_CONTENTS,
+    payload,
+  }),
+  setIsNewArchivedCourse: (payload) => ({
+    type: SET_IS_NEW_ARCHIVED_COURSE,
+    payload,
+  }),
+  updateDismissedArchivedCourse: (payload) => ({
+    type: UPDATE_DISMISSED_ARCHIVED_COURSE,
+    payload,
+  }),
 };
 
 function getHighlightSetsFromState(state) {
   return state.enterpriseCuration?.highlightSets || [];
 }
 
+function getIsNewArchivedCourseFromState(payload) {
+  const dismissedCookies = archivedHighlightsCoursesCookies.get(NEW_ARCHIVED_COURSE_ALERT_DISMISSED_COOKIE_NAME);
+  const isDismissedArchivedCoursesSet = dismissedCookies !== undefined;
+  const dismissedArchivedCourses = isDismissedArchivedCoursesSet ? dismissedCookies : {};
+  let isNewArchivedCourse = false;
+  // checks that the highlight uuid isn't set
+  payload?.forEach(highlightSet => {
+    if (!dismissedArchivedCourses[highlightSet.uuid]) {
+      highlightSet.highlightedContent.forEach(courseContent => {
+        // if the length is greater than 1 for the course run status e.g. ["archived", "published"]
+        // the course should not be archived
+        if (courseContent.courseRunStatuses?.length === 1 && courseContent.courseRunStatuses?.includes('archived')) {
```

Comment:
[sanity check] Just to verify for my own understanding, this feature intends to only work with course content? That is, this feature isn't intended to work for archived courses that may be part of a program and/or pathway (which can also be highlighted by an admin)?

Related to this, I might recommend renaming `courseContent` variable to something else given its value is not always a course (can either be a course, program, or pathway).

Reply:
This is a good callout. We are intending for this to work with pathways and programs as well. @kiram15 merged in a couple of PR's that added course run statuses to pathways and programs.

https://github.com/openedx/course-discovery/pull/4248
https://github.com/openedx/course-discovery/pull/4261